ospfd/ospf_main.c

```diff
@@ -134,6 +134,38 @@ FRR_DAEMON_INFO(ospfd, OSPF, .vty_port = OSPF_VTY_PORT,
 		.n_yang_modules = array_size(ospfd_yang_modules),
 );
 
+/** Max wait time for config to load before accepting hellos */
+#define OSPF_PRE_CONFIG_MAX_WAIT_SECONDS 600
+
+static void ospf_config_finish(struct event *t)
+{
+	zlog_err("OSPF configuration end timer expired after %d seconds.",
+		 OSPF_PRE_CONFIG_MAX_WAIT_SECONDS);
+}
+
+static void ospf_config_start(void)
+{
+	EVENT_OFF(t_ospf_cfg);
+	if (IS_DEBUG_OSPF_EVENT)
+		zlog_debug("ospfd config start callback received.");
+	event_add_timer(master, ospf_config_finish, NULL,
+			OSPF_PRE_CONFIG_MAX_WAIT_SECONDS, &t_ospf_cfg);
+}
+
+static void ospf_config_end(void)
+{
+	if (IS_DEBUG_OSPF_EVENT)
+		zlog_debug("ospfd config end callback received.");
+
+	/* If the OSPF config processing timer thread isn't running, then
+	 * we can safely assume, there is nothing more to do.
+	 */
+	if (!event_is_scheduled(t_ospf_cfg))
```

Comment:
Still, I think this check is not needed. EVENT_OFF should do the same actually.

Reply:
I agree. I'll remove at the risk of having to do CI:rerun several times. I followed IS-IS where you actually save not originating the LSPs.